package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,20 @@
 #!/bin/sh
 
-CFG=$1
+CFG="${1:-/etc/board.json}"
 
-[ -n "$CFG" ] || CFG=/etc/board.json
-
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
-		$(. $a)
+if [ -d '/etc/board.d/' ] && \
+   [ ! -s "${CFG}" ]; then
+	for _script in '/etc/board.d/'*; do
```

Comment:
Why not just `script`? Leading underscore just seems to introduce noise,

Reply:
It depends. Alpine for example (OpenWRT is not alpine, I know :p) makes it mandatory that all 'local' variables are be prefixed with an underscore. Reasoning being from other programming languages where this is done as well, is to identify local/private variables. Since we can't do `for local script` this is the next best thing.

So it doesn't add noise, it helps to identify that this is a intended to be a loop-local private variable, where it's scope shouldn't be considered after the loop (or function in the context of a function, especially if you consider that `local` is not posix compliant sadly).

As for noise, equally in the eye of the beholder, to me it is very clear it is a local private variable :)